src/mock_crypto.rs

```diff
@@ -0,0 +1,242 @@
+                let mut temp = m.to_vec();
+                temp.extend(&pk.0);
+                *signature == Signature(hash256(&temp))
+            }
+
+            fn hash256(data: &[u8]) -> [u8; 32] {
+                use tiny_keccak::sha3_256;
+                sha3_256(data)
+            }
+        }
+
+        /// Mock encryption.
+        pub mod box_ {
+            use super::super::with_rng;
+            use rand::Rng;
+
+            /// Number of bytes in a `PublicKey`.
+            pub const PUBLICKEYBYTES: usize = 32;
+            /// Number of bytes in a `SecretKey`.
+            pub const SECRETKEYBYTES: usize = 32;
+            /// Number of bytes in a `Nonce`.
+            pub const NONCEBYTES: usize = 4;
+
+            /// Mock public key for asymmetric encryption/decryption.
+            #[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd,
+                     Serialize)]
+            pub struct PublicKey(pub [u8; PUBLICKEYBYTES]);
+
+            /// Mock secret key for asymmetric encryption/decryption.
+            #[derive(Clone, Debug, Eq, PartialEq)]
+            pub struct SecretKey(pub [u8; SECRETKEYBYTES]);
+
+            /// Mock nonce for asymmetric encryption/decryption.
+            #[derive(Serialize, Deserialize)]
+            pub struct Nonce(pub [u8; NONCEBYTES]);
+
+            #[derive(Clone, Debug, Eq, PartialEq)]
+            pub struct PrecomputedKey(pub [u8; 32]);
+
+            /// Generate mock public and corresponding secret key.
+            pub fn gen_keypair() -> (PublicKey, SecretKey) {
+                with_rng(|rng| {
+                    let value = rng.gen();
+                    (PublicKey(value), SecretKey(value))
+                })
+            }
+
+            /// Generate mock nonce.
+            pub fn gen_nonce() -> Nonce {
+                with_rng(|rng| Nonce(rng.gen()))
+            }
+
+            /// Generate mock shared key
+            pub fn precompute(pk: &PublicKey, _sk: &SecretKey) -> PrecomputedKey {
+                PrecomputedKey(pk.0)
+            }
+
+            /// Perform mock encryption of the given message using the shared key
+            /// and nonce.
+            pub fn seal_precomputed(m: &[u8], nonce: &Nonce, sk: &PrecomputedKey) -> Vec<u8> {
+                let mut result = Vec::with_capacity(m.len() + nonce.0.len() + sk.0.len());
+                result.extend(&nonce.0);
+                result.extend(&sk.0);
+                result.extend(m);
+                result
+            }
+
+            /// Perform mock decryption of the given ciphertext using their secret key, our public
+            /// key and nonce.
+            pub fn open_precomputed(
+                c: &[u8],
+                nonce: &Nonce,
+                sk: &PrecomputedKey,
+            ) -> Result<Vec<u8>, ()> {
+                let n = nonce.0.len();
+                let s = sk.0.len();
+
+                if c[0..n] != nonce.0 {
```

Comment:
why is this done only for nonce and not for secret-key ?

Reply:
Should be fixed with a new commit.